### ISSUE TITLE
IE9 XMLSerializer serializeToString fails to handle Ajax XML

### ISSUE THREAD

**Issue by rkernahan** [clientside/amplesdk#127]
Built a test case to demonstrate that in IE9 serializeToString works fine on local XML data (embedded in the HTML page) but fails to serialize XML data loaded via Ajax (ie. different documentOwner).

Fails with message "No such interface supported".
IE9 implements the XMLSerializer object and has a member called serializeToString but it is not implemented (gives the error message just mentioned).

If I cannot attach a test case, I'll send it by email.

**Comment by rkernahan** [comment]
testcase.html

``` html
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <title>Test case for XMLSerializer</title>
    <script type="text/javascript" src="runtime-dev.js"></script>
    <script type="text/javascript" src="xhtml.js"></script>
    <style type="text/css">
        #viewLocal, #viewAjax {
            height: 120px;
            width: 400px;
            margin: 10px;
            padding: 2px;
            overflow: auto;
            border: 1px solid #aaa;
        }
    </style>
</head>
<body>
    <h1>Test case for XMLSerializer in IE9</h1>

    <p>Press '<em>Load local data</em>' to load some data from the page itself and convert it to a string.</p>
    <button onclick="loadLocal()">Load local data</button>
    <div id="viewLocal">-- no data --</div>

    <p>Press '<em>Load Ajax data</em>' to serialize XML loaded via Ajax.</p>
    <button onclick="loadAjax()">Load Ajax data</button>
    <div id="viewAjax">-- no data --</div>

    <script type="text/javascript">
        function loadLocal() {
            // get the XML from the document as a string; then parse it to XML so we can exercise serializeToString()
            var sRaw = document.getElementById('t1').innerHTML;
            var dp = new DOMParser();
            var oDoc = dp.parseFromString(sRaw, 'text/xml');
            var oRoot = oDoc.documentElement;
            var oXS = new XMLSerializer();
            var sXML = oXS.serializeToString(oRoot);
            document.getElementById('viewLocal').innerHTML = sXML.replace(/\</g,'&lt;').replace(/\>/g, '&gt;').replace(/\n/g,'<br/>');
        }
        function loadAjax() {
            var that = this;
            var oRequest = {};
            oRequest.type = 'GET';
            oRequest.dataType = 'xml';
            oRequest.url = 'vanuatu.xml';
            oRequest.error = function(oXHR) {
                alert('failed to load XML from vanuatu.xml');
            };
            oRequest.success = function(oDoc, success, oXHR) {
                var oRoot = oDoc.documentElement;
                var oXS = new XMLSerializer();
                var sXML = oXS.serializeToString(oRoot);
                console.log(sXML);
                document.getElementById('viewAjax').innerHTML = sXML.replace(/\</g,'&lt;').replace(/\>/g, '&gt;').replace(/\n/g,'<br/>');
            }
            ample.ajax(oRequest);
        }
    </script>

    <script type="text/xml" id="t1">
        <dataSet>
            <record id='SQT26NWC'>
                <company>Jayme Winters</company>
                <price>80</price>
                <increase>0</increase>
                <state>SQT26NWC</state>
                <email>sit.amet@elitpellentesque.edu</email>
            </record>
            <record id='RJI15OAB' disabled="true">
                <company>Elaine Morrow</company>
                <price>89</price>
                <increase>1</increase>
                <state>RJI15OAB</state>
                <email>at.lacus.Quisque@nonummyac.com</email>
            </record>
            <record id='XWE64AEE'>
                <company>Darrel Obrien</company>
                <price>79</price>
                <increase>0</increase>
                <state>XWE64AEE</state>
                <email>Cras@estacfacilisis.org</email>
            </record>
        </dataSet>
    </script>

</body>
</html>

```

**Comment by rkernahan** [comment]
vanuatu.xml - the data loaded via Ajax

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<dataset xmlns="">
    <record id="db56">
        <name>Reggae</name>
        <age>24</age>
    </record>
    <record id="aw72">
        <name>Motown</name>
        <age>73</age>
    </record>
</dataset>
```

**Comment by finalstep** [comment]
Duplicate of #125

**Comment by ilinsky** [comment]
The problem appear to be caused by Internet Explorer internal inconsistency - it fails running XMLSerializer over its own XMLHttpRequest.responseXML object type. I will apply workaround, though it requires user-agent sniffing.. and it is not known what will be the behavior of IE10.

**Comment by ilinsky** [comment]
Fixed per rev f29445c8786fab517221c8816c1869c3c522cf0b
